packages/ui/src/components/widgets/TokenWidget/TokenWidgetRenderer.tsx

```diff
@@ -0,0 +1,1312 @@
+    currency: toToken?.address ? (toToken.address as Address) : undefined,
+    enabled: toToken !== undefined,
+    refreshInterval: undefined,
+    wallet
+  })
+
+  const invalidateBalanceQueries = useCallback(() => {
+    const invalidatePeriodically = (invalidateFn: () => void) => {
+      let maxRefreshes = 4
+      let refreshCount = 0
+      const timer = setInterval(() => {
+        if (maxRefreshes === refreshCount) {
+          clearInterval(timer)
+          return
+        }
+        refreshCount++
+        invalidateFn()
+      }, 3000)
+    }
+
+    queryClient.invalidateQueries({ queryKey: ['useDuneBalances'] })
+
+    // Dune balances are sometimes stale, because of this we need to aggressively fetch them
+    // for a predetermined period to make sure we get back a fresh response
+    if (fromBalanceIsDune) {
+      invalidatePeriodically(() => {
+        queryClient.invalidateQueries({ queryKey: fromBalanceQueryKey })
+      })
+    } else {
+      queryClient.invalidateQueries({ queryKey: fromBalanceQueryKey })
+    }
+    if (toBalanceIsDune) {
+      invalidatePeriodically(() => {
+        queryClient.invalidateQueries({ queryKey: toBalanceQueryKey })
+      })
+    } else {
+      queryClient.invalidateQueries({ queryKey: toBalanceQueryKey })
+    }
+  }, [
+    queryClient,
+    fromBalanceQueryKey,
+    toBalanceQueryKey,
+    toBalanceIsDune,
+    fromBalanceIsDune,
+    address
+  ])
+  const { data: capabilities } = useCapabilities({
+    query: {
+      enabled:
+        connector &&
+        (connector.id === 'coinbaseWalletSDK' || connector.id === 'coinbase')
+    }
+  })
+  const hasAuxiliaryFundsSupport = Boolean(
+    fromToken?.chainId
+      ? capabilities?.[fromToken?.chainId]?.auxiliaryFunds?.supported
+      : false
+  )
+
+  const isSvmSwap = fromChain?.vmType === 'svm' || toChain?.vmType === 'svm'
+  const isBvmSwap = fromChain?.vmType === 'bvm' || toChain?.vmType === 'bvm'
+  const linkedWallet = linkedWallets?.find(
+    (linkedWallet) =>
+      address ===
+        (linkedWallet.vmType === 'evm'
+          ? linkedWallet.address.toLowerCase()
+          : linkedWallet.address) || linkedWallet.address === address
+  )
+
+  const isRecipientLinked =
+    (recipient
+      ? linkedWallets?.find((wallet) => wallet.address === recipient)
+      : undefined) !== undefined
+
+  const isValidFromAddress = isValidAddress(
+    fromChain?.vmType,
+    address ?? '',
+    fromChain?.id,
+    linkedWallet?.connector,
+    connectorKeyOverrides
+  )
+  const fromAddressWithFallback = addressWithFallback(
+    fromChain?.vmType,
+    address,
+    fromChain?.id,
+    linkedWallet?.connector,
+    connectorKeyOverrides
+  )
+
+  const isValidToAddress = isValidAddress(
+    toChain?.vmType,
+    recipient ?? '',
+    toChain?.id
+  )
+
+  const toAddressWithFallback = addressWithFallback(
+    toChain?.vmType,
+    recipient,
+    toChain?.id
+  )
+
+  const externalLiquiditySupport = useQuote(
```

Comment:
we need to actually drop this external liquidity stuff. Probably easiest to keep it for now and in the future drop it all at once in both widgets. Just wanted to bring this up.